### PR TITLE
commit jquery-croper type definitions

### PR DESCRIPTION
**Pull request by mustafasalahuldin**
commit jquery-croper type definitions

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [x] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/jquery-cropper/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// Type definitions for jquery-cropper v1.0.1
+// Project: https://github.com/davidstutz/jquery-cropper
```
Comment (RyanCavanaugh) [inline types/jquery-cropper/index.d.ts]: This link doesn't work

File: types/jquery-cropper/tslint.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
+{
```
Comment (RyanCavanaugh) [inline types/jquery-cropper/tslint.json]: Do not bulk disable the lint rules.

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]

File: types/jquery-cropper/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    crop?(data?: any): void;
+    zoom?(): any;
+}
+
+interface JQuery {
+    cropper(options?: ICropperOptions): JQuery;
+    cropper(method: string, ...arguments: any[]): JQuery;
+}
+
+declare module "jquery-cropper" {
```
Comment (RyanCavanaugh) [inline types/jquery-cropper/index.d.ts]: What is this doing here?

File: types/jquery-cropper/index.d.ts (hunk 1 above)
Comment (mustafasalahuldin) [reply types/jquery-cropper/index.d.ts]: the correct link is:
https://github.com/fengyuanchen/jquery-cropper

File: types/jquery-cropper/index.d.ts (hunk 3 above)
Comment (mustafasalahuldin) [reply types/jquery-cropper/index.d.ts]: which line are you referring to?

File: types/jquery-cropper/tslint.json (hunk 2 above)
Comment (mustafasalahuldin) [reply types/jquery-cropper/tslint.json]: OK, I'll modify then push again.

File: types/jquery-cropper/index.d.ts (hunk 3 above)
Comment (mustafasalahuldin) [reply types/jquery-cropper/index.d.ts]: I removed the 'declare module "jquery-cropper"' line.

**Comment by mustafasalahuldin** [comment]
Hello
Thanks very much.
I'm just a newbie and I don't know where to look at the new code to update the review status.
Please advise.
Also I resolved all our conversations if that what you are referring to.


My best regards,

Mustafa Salaheldin Ali
 Data Science Manager – Digital Disruption at WS ATKINS & Partners Overseas, Middle East & Africa




ATKINS

Find out more about what we do and how we do it – www.atkinsglobal.com
4th Floor, Al Rostamani Building, Khalid Bin Waleed Road, Bur Dubai, 5620 

Tel: +971 4 4059 300 | Mobile: +971 5 5419 1145

Email: mustafa.ali@atkinsglobal.com  | Web: www.atkinsglobal.com | Careers: www.atkinsglobal.com/careers 
 

    On Sunday, January 5, 2020, 11:24:54 AM GMT+4, TypeScript Bot <notifications@github.com> wrote:  
 
 
 🔔 @RyanCavanaugh - Thanks for your review of this PR! Can you please look at the new code and update your review status if appropriate?

**Review by mustafasalahuldin — COMMENTED** [review]
Final code.

File: types/jquery-cropper/tslint.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "interface-name": false
```
Comment (sheetalkamat) [inline types/jquery-cropper/tslint.json]: Do not disable the rule here. Fix errors or disable the error at the line where you need to explaining why that is correct

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/jquery-cropper/tslint.json (hunk 4 above)
Comment (mustafasalahuldin) [reply types/jquery-cropper/tslint.json]: unfortunately I'm on vacation now, I'll be back next week, then I can send the updates after 15th of March.
I tried to update it last week but it seemed like the PR was already deleted, that is why I wait until I come back from vacation.

**Pull request closed without merging**